themis/spec/secure-cell.md

```diff
@@ -220,3 +220,166 @@ AES-GCM algorithm has a limit on how much data can be safely encrypted using the
+
+    +---------------------------------------------------------------+
+    |                                                               |
+    +                         encrypted data                        +
+    |                                                               |
+    +---------------------------------------------------------------+
+
+### Authentication token: symmetric keys
+
+Layout of the authentication token depends on the type of secret in use.
+For symmetric keys the token looks like this:
+
+     0               4               8               12              16
+    +---------------+---------------+---------------+---------------+
+    |  algorithm ID |   IV length   |auth tag length| message length|
+    +---------------+---------------+---------------+---------------+
+    |                    IV data                    |    auth tag   >
+    +---------------+---------------+---------------+---------------+
+    >                auth tag (cont.)               |
+    +---------------+---------------+---------------+
+
+where
+
+  - **algorithm ID** (4 bytes) describes the encryption algorithm
+  - **IV length** (4 bytes) stores the length of **IV data** in bytes, normally 12
+  - **auth tag length** (4 bytes) stores the length of **auth tag** in bytes, normally 16
+  - **message length** (4 bytes) stores the length of **encrypted data** in bytes
+  - **IV data** (12 bytes) stores the random initialisation vector for encryption
+  - **auth tag** (16 bytes) stores integrity authentication tag produced by encryption
+
+All non-data fields are stored in *little-endian* byte order.
+**IV data** and **auth tag** are interpreted as is.
+
+{{< hint info >}}
+Contrary to most other data structures in Themis,
+Secure Cell uses little-endian due to historical reasons.
+{{< /hint >}}
+
+The **algorithm ID** field is actually a bitmask with
+[Soter symmetric algorithm descriptor](../common/#soter-symmetric-algorithm-descriptor):
+
+          28      24      20      16      12       8       4       0
+    +-------+-------+-------+-------+-------+-------+-------+-------+
+    |  alg. |  KDF  |  ---  |padding|  ---  |      key length       |
+    +-------+-------+-------+-------+-------+-------+-------+-------+
+
+As of Themis 0.13 released in 2020,
+Secure Cell in Seal and Token Protect mode uses AES-GCM, with Soter KDF, and PKCS#7 padding.
+This results in the following descriptors currently being in use for symmetric keys:
+
+| Encryption algorithm       | Algorithm ID |
+| -------------------------- | ------------ |
+| AES-256-GCM **(default)**  | `0x40010100` |
+| AES-196-GCM                | `0x400100C0` |
+| AES-128-GCM (_deprecated_) | `0x40010080` |
+
+Refer to the [algorithm descriptor overview](../common/#soter-symmetric-algorithm-descriptor) for details.
+
+The **IV data** and **auth tag** fields are theoretically flexible,
+but with current algorithm choice IV is always 12 bytes long
```

Comment:
Probably, we should explain why IV is 12 bytes length. It is a recommended parameter (in context of GCM mode) for interoperability, efficiency and security (in some sense). We can reference NIST SP-800 38D document for curious readers

Explanation: 
1) https://crypto.stackexchange.com/questions/41601/aes-gcm-recommended-iv-size-why-12-bytes;
2) https://crypto.stackexchange.com/questions/59396/generate-j-0-for-gcm-cipher-when-operatornameleniv-neq-96-bits

Reply:
> Probably, we should explain why IV is 12 bytes length.

Makes sense. Will do so for PBKDF2 salt length too, I remember referencing some other recommendation for it.